### PR TITLE
Fine tune fitness assignment in supervised learning

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: Core/Sources/Implementations/phenotype/neuron.cpp [hunk 1]
```diff
@@ -46,14 +46,14 @@ auto Neuron::RequestDataAndGetActionPotential() -> Type::neuron_value_t {
...
 	}
-	lastActionPotential = sigmoid(incomingPotentials);
-	return lastActionPotential;
+	return lastActionPotential = (*activationFunction)(incomingPotentials);
 }
 
-auto Neuron::sigmoid(Type::neuron_value_t d) -> Type::neuron_value_t {
-	return tanh(d);
+auto Neuron::GetActivationFunctionUsed()->std::unique_ptr<ActivationFunction::IActivationFunction> {
+    return std::make_unique<ActivationFunction::Tanh>();
```
Comment (jeremystucki) [inline Core/Sources/Implementations/phenotype/neuron.cpp]: https://marketplace.visualstudio.com/items?itemName=EditorConfigTeam.EditorConfig

Comment (janhohenheim) [reply Core/Sources/Implementations/phenotype/neuron.cpp]: image

Comment (janhohenheim) [reply Core/Sources/Implementations/phenotype/neuron.cpp]: image

**Comment by janhohenheim** [comment]
don't delete the branch yet

**Pull request merged**